cern_access/indico_cern_access/tests/operations_test.py

```diff
@@ -0,0 +1,129 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import pytest
+
+from indico.core import signals
+from indico.modules.events.registration.util import create_registration, make_registration_form, modify_registration
+
+from indico_cern_access.models.access_requests import CERNAccessRequestState
+from indico_cern_access.tests.conftest import PERSONAL_DATA
+from indico_cern_access.util import RegformDataMode, grant_access
+
+
+@pytest.fixture
+def api_delete(mocker):
+    """Mock up DELETE request to ADAMS."""
+    mock = mocker.patch('indico_cern_access.plugin.send_adams_delete_request', autospec=True)
+    mocker.patch('indico_cern_access.util.send_adams_delete_request', new=mock)
+    return mock
+
+
+@pytest.fixture
+def api_post(mocker):
+    """Mock up POST request to ADAMS."""
+    def _mock_post(event, registrations, **kwargs):
+        return CERNAccessRequestState.active, {reg.id: {'$rc': 'test'} for reg in registrations}
+
+    mock = mocker.patch('indico_cern_access.plugin.send_adams_post_request', side_effect=_mock_post, autospec=True)
+    mocker.patch('indico_cern_access.util.send_adams_post_request', new=mock)
+    return mock
+
+
+@pytest.fixture
+def dummy_access_request(dummy_regform):
+    """Create a registration and corresponding request."""
+    form = make_registration_form(dummy_regform)(csrf_enabled=False)
+    form.validate()
+    return create_registration(dummy_regform, form.data).cern_access_request
+
+
+def setup_fixtures(func):
+    """Set up fixtures (utlity decorator)."""
+    pytest.mark.usefixtures('smtp', 'mock_access_request', 'dummy_access_request')(func)
```

Comment:
shouldn't there be `func = ...` even in case the func isn't actually replaced with something new?

Reply:
I guess it depends on what the decorator does. This one doesn't wrap the function, from what I see. But yeah, it makes more sense.

Reply:
yeah, just to avoid problems if pytest ever decides to replace the function with something else